src/sortStudents.ts

```diff
@@ -1,16 +1,52 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade'
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function getStudentAverageGrade({ grades }: Student): number {
+  return grades.reduce((sum, grade) => sum + grade, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const signCoefficient = order === 'asc' ? 1 : -1;
+
+  return [...students].sort((studentA: Student, studentB: Student) => {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return signCoefficient
+          * (studentA[sortBy].localeCompare(studentB[sortBy]));
+
+      case SortType.Age:
+      case SortType.Married:
+        return signCoefficient
+          * (Number(studentA[sortBy]) - Number(studentB[sortBy]));
+
+      case SortType.AverageGrade:
+        return signCoefficient
+          * (getStudentAverageGrade(studentA)
+            - getStudentAverageGrade(studentB));
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+      default:
+        throw new Error('Wrong sort type!');
```

Comment:
```suggestion
        throw new Error(`Wrong sort type: ${sortBy}!`);
```
Always try to give a useful description.